File: Algorithm/Recursion/Solution/hanoi - douv/hanoi.c
#include <stdio.h>

void put(int from, int to, int count)
{
    if (count == 1)
    {
        printf("%d -> %d\n", from, to);
        return;
    }

    put(from, 6 - from - to, count - 1);
    put(from, to, 1);
    put(6 - from - to, to, count - 1);
}

int main(void)
{
    int N;
    scanf("%d", &N);
    put(1, 3, N);
    return 0;
}

File: Algorithm/Recursion/Solution/hanoi - douv/hanoi.py
def putDisk(origin, dest, count):
    if count == 1:
        print("%d -> %d" % (origin, dest))
        return

    putDisk(origin, 6-origin-dest, count-1)
    putDisk(origin, dest, 1)
    putDisk(6-origin-dest, dest, count-1)


putDisk(1, 3, int(input()))
